libraries/bot-applicationinsights/src/test/java/com/microsoft/bot/applicationinsights/BotTelemetryClientTests.java

```diff
@@ -0,0 +1,183 @@
+        TelemetryClient telemetryClient = new TelemetryClient(telemetryConfiguration);
+
+        botTelemetryClient = new BotTelemetryClientImpl(telemetryClient);
+    }
+
+    @Test
+    public void NullTelemetryClientThrows() {
+        try {
+            new BotTelemetryClientImpl(null);
+        } catch (Exception e) {
+            Assert.assertEquals("IllegalArgumentException", e.getCause());
+        }
+    }
+
+    @Test
+    public void NonNullTelemetryClientSucceeds() {
+        TelemetryClient telemetryClient = new TelemetryClient();
+
+        BotTelemetryClient botTelemetryClient = new BotTelemetryClientImpl(telemetryClient);
+    }
+
+    @Test
+    public void OverrideTest() {
+        TelemetryClient telemetryClient = new TelemetryClient();
+        MyBotTelemetryClient botTelemetryClient = new MyBotTelemetryClient(telemetryClient);
+    }
+
+    @Test
+    public void TrackAvailabilityTest()
+    {
+        Map<String, String> properties = new HashMap<>();
+        Map<String, Double> metrics = new HashMap<>();
+        properties.put("hello", "value");
+        metrics.put("metric", 0.6);
+
+        botTelemetryClient.trackAvailability(
+            "test",
+            OffsetDateTime.now(),
+            Duration.ofSeconds(1000), // TODO: use computer ticks
+            "run location",
+            true,
+            "message",
+            properties,
+            metrics);
+
+        TelemetryChannel telemetryClient = Mockito.mock(TelemetryChannel.class);
+        Mockito.doAnswer(invocation -> {
+            AvailabilityTelemetry eventName = invocation.getArgument(0);
+
+            Assert.assertEquals("test", eventName.getName());
+            Assert.assertEquals("value", eventName.getProperties().get("hello"));
+            Assert.assertEquals("0.6", eventName.getMetrics().get("metric"));
+
+            return null;
+        }).when(telemetryClient).send(Mockito.any(AvailabilityTelemetry.class));
+    }
+
+    @Test
+    public void TrackEventTest()
+    {
+        botTelemetryClient.trackEvent("test", new HashMap<String, String>() {{ put("hello", "value"); }}, new HashMap<String, Double>() {{ put("metric", 0.6); }});
+
+        TelemetryChannel telemetryClient = Mockito.mock(TelemetryChannel.class);
+        Mockito.doAnswer(invocation -> {
+            AvailabilityTelemetry eventName = invocation.getArgument(0);
+
+            Assert.assertEquals("test", eventName.getName());
+            Assert.assertEquals("value", eventName.getProperties().get("hello"));
+            Assert.assertEquals("0.6", eventName.getMetrics().get("metric"));
+
+            return null;
+        }).when(telemetryClient).send(Mockito.any(AvailabilityTelemetry.class));
+    }
+
+    @Test
+    public void TrackDependencyTest()
+    {
+        botTelemetryClient.trackDependency("test", "target", "dependencyname", "data", OffsetDateTime.now(), Duration.ofSeconds(1000), "result", false);
```

Comment:
```suggestion
        botTelemetryClient.trackDependency(
            "test",
            "target",
            "dependencyname",
            "data",
            OffsetDateTime.now(),
            Duration.ofSeconds(1000),
            "result",
            false);
```

Also, can we add the `//TODO` you added before about the `Duration.ofSeconds(1000)`?